toplevel/vernac.ml

```diff
@@ -145,6 +152,7 @@ let rec interp_vernac sid po (loc,com) =
       let com = if !Flags.time then VernacTime (loc,com) else com in
       interp com
     with reraise ->
+      ignore(Stm.edit_at sid);
```

Comment:
> coqtop does not, so indeed you have to edit_at the last known safe id.

Indeed, I edit at the input id that is required to be valid, I think that is correct.